lib/puppet_metadata/github_actions.rb

```diff
@@ -137,14 +137,34 @@ def puppet_version_below_minimum?(version)
     end
 
     def os_release_to_beaker_setfile(os, release, puppet_collection)
-      PuppetMetadata::Beaker.os_release_to_setfile(
-        os,
-        release,
-        use_fqdn: options[:beaker_use_fqdn],
-        pidfile_workaround: options[:beaker_pidfile_workaround],
-        domain: options[:domain],
-        puppet_version: puppet_collection,
-      )
+      if options[:hosts]
+        setfiles = options[:hosts].collect do |host|
+          PuppetMetadata::Beaker.os_release_to_setfile(
+            os,
+            release,
+            use_fqdn: options[:beaker_use_fqdn],
+            pidfile_workaround: options[:beaker_pidfile_workaround],
+            domain: options[:domain],
+            puppet_version: puppet_collection,
+            hostname: host,
+          )
+        end
+
+        # check for os/release pairs that don't have a setfile
+        return setfiles[0] if setfiles[0].nil?
```

Comment:
In Ruby 2.7 you can use `filter_map` as a short hand for `.map.compact` (`.collect` is an alias for `.map`) which is IMHO results in a bit more descriptive in the return statement.

```suggestion
        setfiles = options[:hosts].filter_map do |host|
          PuppetMetadata::Beaker.os_release_to_setfile(
            os,
            release,
            use_fqdn: options[:beaker_use_fqdn],
            pidfile_workaround: options[:beaker_pidfile_workaround],
            domain: options[:domain],
            puppet_version: puppet_collection,
            hostname: host,
          )
        end

        return if setfiles.empty?
```